src/components/CompareResults/beta/ResultsTable.tsx

```diff
@@ -64,10 +68,25 @@ function processResults(results: CompareResultsItem[]) {
 
 function ResultsTable(props: ResultsTableProps) {
   const { themeMode } = props;
-  const compareResults: CompareResultsItem[] = useAppSelector(
+
+  const resultsData = useAppSelector(
     (state: RootState) => state.compareResults.data,
   );
-  const processedResults = processResults(compareResults);
+
+  const loading = useAppSelector(
+    (state: RootState) => state.compareResults.loading,
+  );
+
+  const [compareData, setCompareData] = useState<CompareResultsItem[]>([]);
+
+  useEffect(() => {
+    const data = [].concat(...Object.values(resultsData)); // concatenate all results
+    if (Object.keys(resultsData).length > 0) {
+      setCompareData(data);
+    }
+  }, [resultsData]);
+
```

Comment:
I will not add the code snippet to `processResults` because I want to be able to also display results from a selected revision, if I add this logic then I'll only display the results from all the revisions. I removed the `useEffect()`.

Reply:
as long as there's no state + effect, I'm fine. In the future we'll move this selector to a memoized selector anyway, so we'll be able to revisit.